// code derived from https://github.com/vdemedes/secure-compare

// https://github.com/vdemedes/secure-compare#license
/*
secure-compare is released under the MIT license.
*/

diff --git a/index.js b/index.js
index 1cfd91b..1ffa759 100755
--- a/index.js
+++ b/index.js
@@ -18,8 +18,8 @@ function compare (a, b) {
   }
   
   for (var i = 0, il = a.length; i < il; ++i) {
-    mismatch |= (a.charCodeAt(i) ^ a.charCodeAt(i));
+    mismatch |= (a.charCodeAt(i) ^ b.charCodeAt(i));
   }
   
   return mismatch === 0;
-};
\ No newline at end of file
+};
